multiversx_sdk/network_providers/transaction_decoder.py

```diff
@@ -0,0 +1,236 @@
+
+        esdt_metadata = self.get_esdt_transaction_metadata(metadata)
+        if esdt_metadata:
+            return esdt_metadata
+
+        nft_metadata = self.get_nft_transfer_metadata(metadata)
+        if nft_metadata:
+            return nft_metadata
+
+        multi_metadata = self.get_multi_transfer_metadata(metadata)
+        if multi_metadata:
+            return multi_metadata
+
+        return metadata
+
+    def get_normal_transaction_metadata(self, transaction: ITransactionToDecode) -> TransactionMetadata:
+        metadata = TransactionMetadata()
+        metadata.sender = transaction.sender.to_bech32()
+        metadata.receiver = transaction.receiver.to_bech32()
+        metadata.value = transaction.value
+
+        if transaction.data:
+            data_components = transaction.data.split("@")
+
+            args = data_components[1:]
+            if all(self.is_smart_contract_call_argument(x) for x in args):
+                metadata.function_name = data_components[0]
+                metadata.function_args = args
+
+        return metadata
+
+    def get_esdt_transaction_metadata(self, metadata: TransactionMetadata) -> Optional[TransactionMetadata]:
+        if metadata.function_name != "ESDTTransfer":
+            return None
+
+        args = metadata.function_args
+        if not args:
+            return None
+
+        if len(args) < 2:
+            return None
+
+        identifier = self.hex_to_string(args[0])
+        value = self.hex_to_number(args[1])
+
+        result = TransactionMetadata()
+
+        result.sender = metadata.sender
+        result.receiver = metadata.receiver
+        result.value = value
+        result.transfers = []
+
+        if len(args) > 2:
+            result.function_name = self.hex_to_string(args[2])
+            result.function_args = args[3:]
+
+        token = Token(identifier)
+        transfer = TokenTransfer(token, value)
+
+        result.transfers.append(transfer)
+        return result
+
+    def get_nft_transfer_metadata(self, metadata: TransactionMetadata) -> Optional[TransactionMetadata]:
+        if metadata.sender != metadata.receiver:
+            return None
+
+        if metadata.function_name != "ESDTNFTTransfer":
+            return None
+
+        args = metadata.function_args
+        if not args:
+            return None
+
+        if len(args) < 4:
+            return None
+
+        if not self.is_address_valid(args[3]):
+            return None
+
+        collection_identifier = self.hex_to_string(args[0])
+        nonce = args[1]
+        value = self.hex_to_number(args[2])
+        receiver = Address.new_from_hex(args[3], DEFAULT_HRP)
+
+        result = TransactionMetadata()
+        result.sender = metadata.sender
+        result.receiver = receiver.to_bech32()
+        result.value = value
+        result.transfers = []
+
+        if len(args) > 4:
+            result.function_name = self.hex_to_string(args[4])
+            result.function_args = args[5:]
```

Comment:
the length can be exactly 5 and args[5] is invalid

Reply:
`args[5:]` returns an empty array if the length is `5`.